subprojects/logging/src/main/java/org/gradle/internal/logging/sink/GroupingProgressLogEventGenerator.java

```diff
@@ -84,6 +103,19 @@ private void onStart(ProgressStartEvent startEvent) {
             if (isGroupedOperation(startEvent.getBuildOperationCategory())) {
                 String header = startEvent.getLoggingHeader() != null ? startEvent.getLoggingHeader() : startEvent.getDescription();
                 operationsInProgress.put(buildOpId, new OperationGroup(startEvent.getCategory(), header, startEvent.getTimestamp(), startEvent.getBuildOperationId()));
+
+                if (future == null || future.isCancelled()) {
+                    future = executor.scheduleWithFixedDelay(new Runnable() {
```

Comment:
Might be easier to easier to use `scheduleAtFixedRate` instead of keeping track of the last update time.